locator/tablets.cc

```diff
@@ -66,29 +66,42 @@ read_replica_set_selector get_selector_for_reads(tablet_transition_stage stage)
 }
 
 tablet_transition_info::tablet_transition_info(tablet_transition_stage stage,
+                                               tablet_transition_kind transition,
                                                tablet_replica_set next,
                                                tablet_replica pending_replica,
                                                service::session_id session_id)
     : stage(stage)
+    , transition(transition)
     , next(std::move(next))
     , pending_replica(std::move(pending_replica))
     , session_id(session_id)
     , writes(get_selector_for_writes(stage))
     , reads(get_selector_for_reads(stage))
 { }
 
+tablet_migration_streaming_info get_migration_streaming_info(const tablet_info& tinfo, const tablet_migration_info& trinfo) {
+    return get_migration_streaming_info(tinfo, migration_to_transition_info(tinfo, trinfo));
+}
+
 tablet_migration_streaming_info get_migration_streaming_info(const tablet_info& tinfo, const tablet_transition_info& trinfo) {
-    tablet_migration_streaming_info result = {
-        .read_from = std::unordered_set<tablet_replica>(tinfo.replicas.begin(), tinfo.replicas.end()),
-        .written_to = std::unordered_set<tablet_replica>(trinfo.next.begin(), trinfo.next.end())
-    };
-    for (auto&& r : trinfo.next) {
-        result.read_from.erase(r);
-    }
-    for (auto&& r : tinfo.replicas) {
-        result.written_to.erase(r);
+    tablet_migration_streaming_info result;
+    switch (trinfo.transition) {
+        case tablet_transition_kind::migration:
+            result.read_from = std::unordered_set<tablet_replica>(tinfo.replicas.begin(), tinfo.replicas.end());
+            result.written_to = std::unordered_set<tablet_replica>(trinfo.next.begin(), trinfo.next.end());
+            for (auto&& r : trinfo.next) {
+                result.read_from.erase(r);
+            }
+            for (auto&& r : tinfo.replicas) {
+                result.written_to.erase(r);
+            }
+            return result;
+        case tablet_transition_kind::rebuild:
+            result.written_to = std::unordered_set<tablet_replica>(trinfo.next.begin(), trinfo.next.end());
+            result.read_from = result.written_to;
```

Comment:
I think this is inaccurate. If we consider using repair for the tablet replica
'rebuild', not all new replicas are written to. Only the pending node is being
written to. And the pending node, which has no data, is not being read from.

Reply:
Yes, it's conservative (reserves more than necessary). 

If we use repair, not all replicas may be written too, but we don't know which ones will be. So prefer to avoid overload.

Reply:
I'll make it more accurate.